pkg/postage/batchservice/batchservice_test.go

```diff
@@ -0,0 +1,272 @@
+			t,
+			mock.WithPutErr(testErr, 1),
+		)
+		putBatch(t, batchStore, testBatch)
+
+		if err := svc.TopUp(testBatch.ID, testTopUpAmount); err == nil {
+			t.Fatal("expected error")
+		}
+	})
+
+	t.Run("passes", func(t *testing.T) {
+		svc, batchStore := testNewStoreAndService(t)
+		putBatch(t, batchStore, testBatch)
+
+		want := testBatch.Value
+		want.Add(want, testTopUpAmount)
+
+		if err := svc.TopUp(testBatch.ID, testTopUpAmount); err != nil {
+			t.Fatalf("top up: %v", err)
+		}
+
+		got, err := batchStore.Get(testBatch.ID)
+		if err != nil {
+			t.Fatalf("batch store get: %v", err)
+		}
+
+		if got.Value.Cmp(want) != 0 {
+			t.Fatalf("topped up amount: got %v, want %v", got.Value, want)
+		}
+
+	})
+}
+
+func TestBatchServiceUpdateDepth(t *testing.T) {
+	const testNewDepth = 30
+	testBatch := postagetesting.MustNewBatch()
+
+	t.Run("expect get error", func(t *testing.T) {
+		svc, _ := testNewStoreAndService(
+			t,
+			mock.WithGetErr(testErr, 1),
+		)
+
+		if err := svc.UpdateDepth(testBatch.ID, testNewDepth); err == nil {
+			t.Fatal("expected get error")
+		}
+	})
+
+	t.Run("expect put error", func(t *testing.T) {
+		svc, batchStore := testNewStoreAndService(
+			t,
+			mock.WithPutErr(testErr, 1),
+		)
+		putBatch(t, batchStore, testBatch)
+
+		if err := svc.UpdateDepth(testBatch.ID, testNewDepth); err == nil {
+			t.Fatal("expected put error")
+		}
+	})
+
+	t.Run("passes", func(t *testing.T) {
+		svc, batchStore := testNewStoreAndService(t)
+		putBatch(t, batchStore, testBatch)
+
+		if err := svc.UpdateDepth(testBatch.ID, testNewDepth); err != nil {
+			t.Fatalf("update depth: %v", err)
+		}
+
+		val, err := batchStore.Get(testBatch.ID)
+		if err != nil {
+			t.Fatalf("batch store get: %v", err)
+		}
+
+		if val.Depth != testNewDepth {
+			t.Fatalf("wrong batch depth set: want %v, got %v", testNewDepth, val.Depth)
+		}
+	})
+}
+
+func TestBatchServiceUpdatePrice(t *testing.T) {
+	testChainState := postagetesting.NewChainState()
+	testChainState.Price = big.NewInt(100000)
+	testNewPrice := big.NewInt(20000000)
+
+	t.Run("expect put error", func(t *testing.T) {
+		svc, batchStore := testNewStoreAndService(
+			t,
+			mock.WithChainState(testChainState),
+			mock.WithPutErr(testErr, 1),
+		)
+		putChainState(t, batchStore, testChainState)
+
+		if err := svc.UpdatePrice(testNewPrice); err == nil {
+			t.Fatal("expected error")
+		}
+	})
+
+	t.Run("passes", func(t *testing.T) {
+		svc, batchStore := testNewStoreAndService(
+			t,
+			mock.WithChainState(testChainState),
+		)
+
+		if err := svc.UpdatePrice(testNewPrice); err != nil {
+			t.Fatalf("update price: %v", err)
+		}
+
+		cs, err := batchStore.GetChainState()
+		if err != nil {
+			t.Fatalf("batch store get chain state: %v", err)
+		}
+
+		if cs.Price.Cmp(testNewPrice) != 0 {
+			t.Fatalf("bad price: want %v, got %v", cs.Price, testNewPrice)
+		}
+	})
+}
+
+func testNewStoreAndService(t *testing.T, opts ...mock.Option) (postage.EventUpdater, postage.Storer) {
```

Comment:
the `test` prefix might infer that you're actually testing something. maybe `newTestStore`?

Reply:
Renamed.